### PR TITLE
Implement project admin commands. Fixes #8

### LINKED ISSUE #8: Implement project administration commands
CLI implementation of:
* create-project
* list-projects
* show-project
* drop-project

With unit tests, naturally

### PR DESCRIPTION
**Pull request by tvirolai**
A basic implementation for project administration commands. The `annif init` command also now checks for orphaned indexes and removes them if necessary.

### CODE REVIEW

**Comment by tvirolai** [comment]
Very true. I'll look into this.

**Comment by tvirolai** [comment]
Now `show-project` produces output formatted like this:

`(annif_env) ➜  annif git:(issue8) annif show-project herttilei
details about project 'herttilei':

project name:        herttilei
project language:  en
project analyzer:   english
index:                    annif
score:                    0.2876821
`
`list-projects` works like this:

`(annif_env) ➜  annif git:(issue8) annif list-projects                                               
Currently available projects:

1. herttilei (language: en, analyzer: english)
2. jaahas (language: armenian, analyzer: armenia)`

What do you think?

**Comment by osma** [comment]
Much better now. In `show-project` output, `score` is completely useless and `index` is probably redundant too.

**Comment by tvirolai** [comment]
Ok, I'll remove 'em.

**Comment by osma** [comment]
How about this mocked up output style:

```
$ annif list-projects
Project ID     Language   Analyzer
----------------------------------
herttilei      en         english
jaahas         armenian   armenia
$ annif show-project herttilei
Project ID: herttilei
Language:   en
Analyzer:   english
```

**Comment by tvirolai** [comment]
I also tweaked the test so that running them doesn't affect any other indices that may be in the database.

**Comment by tvirolai** [comment]
Ok, looks nice. I'll update.

**Comment by osma** [comment]
`list-projects` now fails for me with this error:

```
  File "/home/ozone/git/Annif/annif/annif.py", line 87, in listprojects
    parsed += parseRow(p['name'], p['language'], p['analyzer'])
  File "/home/ozone/git/Annif/annif/annif.py", line 81, in parseRow
    return "{0: <15}{1: <15}{2: <15}\n".format(col1, col2, col3)
TypeError: non-empty format string passed to object.__format__
```

**Comment by tvirolai** [comment]
Weird. The only way I managed to reproduce this error was to first create a project with insufficient arguments (e.g. provide only a name but not language or analyzer information) and then try to list the available projects.

Creating projects with insufficient arguments is no longer possible but it seems that the code analyzers now deem the createProject function too complex. I'll refactor.

**Comment by osma** [comment]
That's probably the reason. I had created the project without specifying all the parameters.

File: annif/annif.py [hunk 1]
```diff
@@ -137,15 +139,24 @@ def createProject(projectid, language, analyzer):
     """
     proj_indexname = parseIndexname(projectid)
 
-    # Create an index for the project
-    index.create(index=proj_indexname)
+    if not projectid or not language or not analyzer:
+        print('Usage: annif create-project <projectId> --language <lang>'
```
Comment (osma) [inline annif/annif.py]: Missing a space after `<lang>`

File: annif/annif.py [hunk 2]
```diff
@@ -62,9 +76,19 @@ def listprojects():
     doc = {'size': 1000, 'query': {'match_all': {}}}
     results = es.search(index=annif.config['INDEX_NAME'], doc_type='project',
                         body=doc)
+
     projects = [x['_source'] for x in results['hits']['hits']]
+
+    def parseRow(col1, col2, col3):
```
Comment (osma) [inline annif/annif.py]: Nitpick on naming: I would call this function `formatRow`, not `parseRow`. What it does is the opposite of parsing...

File: annif/annif.py [hunk 3]
```diff
@@ -32,10 +33,15 @@
         }
 
 
-def parseIndexname(projectid):
+def parse_indexname(projectid):
```
Comment (osma) [inline annif/annif.py]: shouldn't this be `format_indexname` or even better `format_index_name`?

File: tests/test_annif.py [hunk 4]
```diff
@@ -25,31 +30,51 @@ def test_start():
...
+    assert not index.exists(annif.parse_indexname(FAILED_PROJECT))
+    result = runner.invoke(annif.create_project,
+                           [FAILED_PROJECT, '--language', 'en'])
+    assert not index.exists(annif.parse_indexname(FAILED_PROJECT))
+    result = runner.invoke(annif.create_project,
+                           [FAILED_PROJECT, '--analyzer', 'english'])
+    assert not index.exists(annif.parse_indexname(FAILED_PROJECT))
+
+
+def test_showproject():
```
Comment (osma) [inline tests/test_annif.py]: rename to `test_show_project`?

File: tests/test_annif.py [hunk 5]
```diff
@@ -25,31 +30,51 @@ def test_start():
...
+    result = runner.invoke(annif.show_project, [TEMP_PROJECT])
+    assert result.exit_code == 0
+    # Test should not fail even if the user queries for a non-existent project.
+    failed_result = runner.invoke(annif.show_project, ['nonexistent'])
+    assert failed_result.exit_code == 0
 
 
-def test_dropProject():
-    result = runner.invoke(annif.dropProject, [TEMP_PROJECT])
+def test_dropproject():
```
Comment (osma) [inline tests/test_annif.py]: ditto, 'test_drop_project`

File: tests/test_annif.py [hunk 6]
```diff
@@ -25,31 +30,51 @@ def test_start():
 
 def test_init():
     result = runner.invoke(annif.init)
-    assert index.exists('annif')
+    assert index.exists(annif.annif.config['INDEX_NAME'])
     assert result.exit_code == 0
 
 
 def test_listprojects():
```
Comment (osma) [inline tests/test_annif.py]: rename to `test_list_projects`?

File: annif/annif.py (hunk 3 above)
Comment (tvirolai) [reply annif/annif.py]: Yep, fixing.

File: tests/test_annif.py (hunk 4 above)
Comment (tvirolai) [reply tests/test_annif.py]: Oh yes.

File: tests/test_annif.py (hunk 5 above)
Comment (tvirolai) [reply tests/test_annif.py]: Fixed in the latest commit.

File: tests/test_annif.py (hunk 6 above)
Comment (tvirolai) [reply tests/test_annif.py]: Fixed in the latest commit.

**Pull request merged**